libr/io/plugin.c

```diff
@@ -158,3 +162,105 @@ R_API int r_io_plugin_write_at(RIODesc *desc, ut64 addr, const ut8 *buf, int len
 	}
 	return 0;
 }
+
+R_API char *r_io_plugin_get_uri_prefix (RIOPlugin *plugin) {
```

Comment:
wtf? why is this necessary? why is this hardcoded?

Reply:
return type should be const char *

Reply:
All the plugin prefixes are hardcoded in r_io_ar_plugin_open() method like
```
static bool r_io_ar_plugin_open(RIO *io, const char *file, bool many) {
	return !strncmp ("ar://", file, 5) || !strncmp ("lib://", file, 6);
}
````
and there are no internal fields that specify the prefix
I made another pr by adding an internal field inside the RIOPlugin and modifying 34 files. Since you told me not to modify internal fields a couple of times. I closed that PR and changed to this.

The PR in question is #13012. 

Thanks for taking the time to review. If both my ways are wrong. Feel free to close this issue with a message pointing the right direction.